scipy/interpolate/_bsplines.py

```diff
@@ -297,6 +298,72 @@ def basis_element(cls, t, extrapolate=True):
         c = np.zeros_like(t)
         c[k] = 1.
         return cls.construct_fast(t, c, k, extrapolate)
+    
+    @classmethod
+    def design_matrix(cls, x, t, k):
+        """
+        Returns a design matrix for BSpline object in "dense" or "CSR" formats.
+        
+        Parameters
+        ----------
+        x : array_like, shape (n,)
+            Points to evaluate the spline at.
+        t : array_like, shape (n,)
+            Sorted 1D array of knots.
+        k : int
+            B-spline degree.
+ 
+        Returns
+        -------
+        Sparse matrix in CSR format.
+
+        Examples
+        --------
+        Construct a design matrix for a B-spline
+        >>> from scipy.interpolate import make_interp_spline, BSpline
+        >>> x = np.linspace(0, np.pi * 2, 4)
+        >>> y = np.sin(x)
+        >>> k = 3
+        >>> bspl = make_interp_spline(x, y, k=k)
+        >>> design_matrix = bspl.design_matrix(x, bspl.t, k)
+        >>> design_matrix.toarray()
+        [[1.        , 0.        , 0.        , 0.        ],
+        [0.2962963 , 0.44444444, 0.22222222, 0.03703704],
+        [0.03703704, 0.22222222, 0.44444444, 0.2962963 ],
+        [0.        , 0.        , 0.        , 1.        ]]
+
+        Construct a design matrix for some vector of knots
+        >>> k = 2
+        >>> t = [-1, 0, 1, 2, 3, 4, 5, 6]
+        >>> x = [1, 2, 3, 4]
+        >>> design_matrix = BSpline.design_matrix(x, t, k).toarray()
+        >>> design_matrix
+        [[0.5, 0.5, 0. , 0. , 0. ],
+        [0. , 0.5, 0.5, 0. , 0. ],
+        [0. , 0. , 0.5, 0.5, 0. ],
+        [0. , 0. , 0. , 0.5, 0.5]]
+
+        Comparing with method introduced in gh-6730
```

Comment:
No need to reference the gh issue here. Just say that the result is equivalent, but constructed in the sparse format